EricGray/cardsFactory.js

```diff
@@ -0,0 +1,184 @@
+console.log("Beginning of Question 3, card constructor redux")
+
+var Card = (function(){
+
+	function Card(id){
+		this.id = id;
+	}
+
+	var rankNames = ['','Ace','Two','Three','Four','Five','Six','Seven','Eight','Nine','Ten',
+                        'Jack','Queen','King'];
+
+	var suitNames = ['','Hearts','Diamonds','Spades','Clubs'];
+
+	var fullSet = [];
+
+		for (var id=0; id<52; ++id) {
+    		fullSet.push(new Card(id));
+    }
+	
+
+	Card.prototype.rank = function(){
+		return Math.floor(this.id/4)+1;
+	}
+
+	Card.prototype.suit = function(){
+		return (this.id%4)+1;
+	}
+
+	Card.prototype.color = function(){
+        var suit=this.suit();
+        return suit && ((suit<3)? "red": "black"); 
+	}
+
+	Card.prototype.name = function(){
+        var rank = this.rank();
+        var suit = this.suit();
+        return rank && suit && (rankNames[rank]+' of '+suitNames[suit]);
+	}
+
+	Card.prototype.isValid = function(num){ // Returns--> true, false
+        return ((typeof num)==="number") //correct type
+                && (num%1 === 0)        //integer
+                && num>=0 && num<=51;   //in range
```

Comment:
The range here is why some of your Tarot cards are showing up as invalid. the upper limit of your range will depend on the number of cards for a given constructor.